nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/trash/test/AbstractTestTrashService.java

```diff
@@ -417,4 +418,18 @@ public void testFollowTransitionBackwardCompatibility() {
 
     }
 
+    @Test
+    public void getCanEmptyTrash() {
+        createDocuments();
+        trashService.trashDocument(doc1);
+        transactionalFeature.nextTransaction();
+
+        assertTrue(trashService.getCanEmptyTrash(fold, principal));
+
+        trashService.trashDocument(fold);
+        transactionalFeature.nextTransaction();
+
+        assertTrue(trashService.getCanEmptyTrash(fold, principal));
```

Comment:
what you want to test here:

- that you can pure the folder children 
- or purge folder and children

i ask this question depending on you answer the java doc should be update / reworked on `getCanEmptyTrash` method ...